src/core/lib/security/credentials/ssl/ssl_credentials.cc

```diff
@@ -119,6 +177,45 @@ void grpc_ssl_credentials::set_max_tls_version(
   config_.max_tls_version = max_tls_version;
 }
 
+grpc_security_status grpc_ssl_credentials::InitializeClientHandshakerFactory(
+    const grpc_ssl_config* config, const char* pem_root_certs,
+    const tsi_ssl_root_certs_store* root_store,
+    tsi_ssl_session_cache* ssl_session_cache,
+    tsi_ssl_client_handshaker_factory** handshaker_factory) {
+  // This class level factory can't have a session cache by design. If we want
```

Comment:
Should this if statement be removed and we can make this into a static function?

Reply:
Friendly ping :)

Reply:
I don't think so - this is the bifurcation. The class level factory can't have a session cache, so if the `ssl_session_cache != nullptr` then we need to make a separate factory with a separate CTX that is not associated with the credentials class.

But in the case that there is _not_ a session cache, we do want to use a factory on the class instance that is only initialized once.

Reply:
So this if statement is essentially saying
"If we don't want to have a session cache, and we have an existing `client_handshaker_factory_` (i.e. `client_handshaker_factory_ != nullptr`), then everything is ok and we can use that factory that is already initialized.